File: src/ros2_demo_py/ros2_demo_py/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

# create class to excute
class SmartPhoneNode(Node):
    def __init__(self):
        self.counter_ = 0
        super().__init__("smartphone")

        self.get_logger().info("SmartPhone has been started")
        self.subcriber_ = self.create_subscription(String, "News", 
                                                   self.message_callback, 10)

    def message_callback(self, msg):
        self.get_logger().info(msg.data)

def main(args = None):
    rclpy.init(args=args)
    node = SmartPhoneNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/ros2_demo_cpp/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
// #include "rclcpp/timer.hpp"

class SmartPhoneNode: public rclcpp::Node
{
public:
    SmartPhoneNode() : Node("smartphone"){
        subcriber_ = this->create_subscription<example_interfaces::msg::String>("robot_news", 10,
                                    std::bind(&SmartPhoneNode::news_callback, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Smartphone has been started");
    }

private:
    void news_callback(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subcriber_;
};


int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartPhoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}